service_container/3.3-di-changes.rst

```diff
@@ -577,6 +577,13 @@ Start by updating the service ids to class names:
     account when looking up for services.
     Furthermore it is always recommended to check for definition existence
     using ``has()`` function.
+    
+.. note::
+
+    If you get rid of deprecations and extend `AbstractController` instead of `Controller` for
+    your controllers, you can skip the rest of this step as `AbstractController` won't provide 
+    a containe where you can get the services directly. All services need to be injected as 
+    explained in `Step 5 Cleanup`_
```

Comment:
Instead of a link to an absolute URL, we can link to a specific part of this article. First, change the link of this line by this:

```rst
explained in the :ref:`step 5 of this article <step-5>`.
```

Then, in the line 718 of this article, add the following reference before `Step 5) Cleanup!`:

```rst
.. _step-5:

Step 5) Cleanup!
~~~~~~~~~~~~~~~~

...
```

Finally, you can remove this link at the bottom of the article: `.. _`Step 5 Cleanup`: https://github.com/symfony/symfony-docs/blob/master/service_container/3.3-di-changes.rst#step-5-cleanup`